examples/rng/device/uniform.cpp

```diff
@@ -0,0 +1,212 @@
+*       numbers using unifrom distribution on a SYCL device (CPU, GPU).
+*
+*******************************************************************************/
+
+// stl includes
+#include <iostream>
+#include <vector>
+
+// oneMKL/SYCL includes
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/rng/device.hpp"
+
+#include "rng_example_helper.hpp"
+
+bool isDoubleSupported(sycl::device my_dev) {
+    return my_dev.get_info<sycl::info::device::double_fp_config>().size() != 0;
+}
+
+// example parameters
+constexpr int seed = 777;
+constexpr std::size_t n = 1024;
+constexpr int n_print = 10;
+
+//
+// examples show usage of rng device functionality, which can be called from both
+// host and device sides with scalar and vector generation
+//
+template <typename Type, std::int32_t VecSize>
+int run_example(sycl::queue& queue) {
+    if (VecSize == 1) {
+        std::cout << "\tRunning scalar example" << std::endl;
+    }
+    else {
+        std::cout << "\tRunning vector example with " << VecSize << " vector size" << std::endl;
+    }
+    // prepare array for random numbers
+    std::vector<Type> r_dev(n), r_host(n);
+
+    // submit a kernel to generate on device
+    {
+        sycl::buffer<Type> r_buf(r_dev.data(), r_dev.size());
+
+        try {
+            queue.submit([&](sycl::handler& cgh) {
+                sycl::accessor r_acc(r_buf, cgh, sycl::write_only);
+                cgh.parallel_for(sycl::range<1>(n / VecSize), [=](sycl::item<1> item) {
+                    size_t item_id = item.get_id(0);
+                    oneapi::mkl::rng::device::mcg59<VecSize> engine(seed, item_id * VecSize);
+                    oneapi::mkl::rng::device::uniform<Type> distr;
+
+                    auto res = oneapi::mkl::rng::device::generate(distr, engine);
+                    if constexpr (VecSize == 1) {
+                        r_acc[item_id] = res;
+                    }
+                    else {
+                        res.store(item_id, get_multi_ptr(r_acc));
+                    }
+                });
+            });
+            queue.wait_and_throw();
+        }
+        catch (sycl::exception const& e) {
+            std::cout << "\t\tSYCL exception\n" << e.what() << std::endl;
+            return 1;
+        }
+
+        std::cout << "\t\tOutput of generator:" << std::endl;
+
+        auto r_acc = sycl::host_accessor(r_buf, sycl::read_only);
+        std::cout << "first " << n_print << " numbers of " << n << ": " << std::endl;
+        for (int i = 0; i < n_print; i++) {
+            std::cout << r_acc[i] << " ";
+        }
+        std::cout << std::endl;
+    } // buffer life-time ends
+
+    // compare results with host-side generation
+    oneapi::mkl::rng::device::mcg59<1> engine(seed);
+    oneapi::mkl::rng::device::uniform<Type> distr;
+
+    int err = 0;
+    for (int i = 0; i < n; i++) {
+        r_host[i] = oneapi::mkl::rng::device::generate(distr, engine);
```

Comment:
Do we need to save the result? Can we just generate a single number and compare it?

Reply:
We can. Fixed